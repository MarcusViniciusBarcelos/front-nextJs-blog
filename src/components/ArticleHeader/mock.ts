import { ArticleHeaderProps } from '.';

// Mock dos dados fictícios
const mockArticleHeaderData: ArticleHeaderProps = {
  id: '1',
  title: 'Título do Artigo',
  excerpt: 'Excerto do artigo...',
  cover: {
    id: '1',
    alternativeText: 'Texto alternativo da capa',
    url: 'https://napratica.org.br/wp-content/uploads/2018/08/tecnica-feynman.jpg',
  },
  author: {
    id: '1',
    displayName: 'Autor do Artigo',
    slug: 'autor-do-artigo',
  },
  categories: [
    {
      id: '1',
      displayName: 'Categoria do Artigo',
      slug: 'categoria-do-artigo',
    },
  ],
  createdAt: '2022-02-18',
};

export default mockArticleHeaderData;
