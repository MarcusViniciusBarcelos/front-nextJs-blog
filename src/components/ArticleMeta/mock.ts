import { ArticleMetaProps } from '.';
import { data } from '../../api/dados.json';

const postData = data.posts.data[1];

const mockArticleMetaData: ArticleMetaProps = {
  createdAt: postData.attributes.createdAt,
  author: {
    id: postData.attributes.author.data.id,
    attributes: {
      slug: postData.attributes.author.data.attributes.slug,
      displayName: postData.attributes.author.data.attributes.displayName,
    },
  },
  categories: postData.attributes.categories.data.map((category) => ({
    id: category.id,
    attributes: {
      slug: category.attributes.slug,
      displayName: category.attributes.displayName,
    },
  })),
};

export default mockArticleMetaData;
