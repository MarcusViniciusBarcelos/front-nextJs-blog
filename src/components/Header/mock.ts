import { HeaderProps } from '.';

export default {
  blogName: 'Marcus Barcelos',
  blogDescription:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud.',
  logo: 'https://res.cloudinary.com/dm5kvicsz/image/upload/v1708022537/Whats_App_Image_2023_07_13_at_15_32_56_000d5b42b2.jpg',
  showText: true,
} as HeaderProps;
