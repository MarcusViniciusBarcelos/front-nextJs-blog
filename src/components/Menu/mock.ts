import { MenuProps } from '.';
import { StrapiImage } from '../../shared-types/strapi-image';

export default {
  links: [
    {
      id: 'link-1',
      link: '/home/1',
      newTab: false,
      text: 'My jobs',
    },
    {
      id: 'link-2',
      link: '/home/2',
      newTab: false,
      text: 'About me',
    },
    {
      id: 'link-3',
      link: '/home/3',
      newTab: false,
      text: 'Services',
    },
    {
      id: 'link-4',
      link: 'http://localhost/home/4',
      newTab: false,
      text: 'External Link',
    },
    {
      id: 'link-5',
      link: '/home/5',
      newTab: false,
      text: 'JavaScript',
    },
    {
      id: 'link-6',
      link: '/home/6',
      newTab: false,
      text: 'Python',
    },
    {
      id: 'link-7',
      link: '/home/7',
      newTab: false,
      text: 'Consectetur',
    },
    {
      id: 'link-8',
      link: '/home/8',
      newTab: false,
      text: 'Text test',
    },
    {
      id: 'link-9',
      link: '/home/9',
      newTab: false,
      text: 'Pac-man',
    },
    {
      id: 'link-10',
      link: '/home/10',
      newTab: false,
      text: 'Star Wars',
    },
    {
      id: 'link-11',
      link: '/home/11',
      newTab: false,
      text: 'A menu',
    },
  ],
  blogName: 'Marcus Barcelos',
  logo: {
    id: 'logo-id', // Add the missing 'id' property
    alternativeText: 'Logo', // Add the missing 'alternativeText' property
    url: 'https://res.cloudinary.com/dm5kvicsz/image/upload/v1708022537/Whats_App_Image_2023_07_13_at_15_32_56_000d5b42b2.jpg', // Add the missing 'url' property
  } as StrapiImage,
} as MenuProps;
