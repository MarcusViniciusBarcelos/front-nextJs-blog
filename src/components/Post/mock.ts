import { PostProps } from '.';
import { data } from '../../api/dados.json';

const postData = data.posts.data[1].attributes;

const mockPostData: PostProps = {
  id: data.posts.data[0].id,
  title: postData.title,
  excerpt: postData.excerpt,
  cover: {
    id: postData.cover.data.id,
    alternativeText: postData.cover.data.attributes.alternativeText,
    url: postData.cover.data.attributes.url,
  },
  content: postData.content,
  createdAt: data.posts.data[0].createdAt,
  author: {
    id: postData.author.data.id,
    displayName: postData.author.data.attributes.displayName,
    slug: postData.author.data.attributes.slug,
  },
  categories: [
    {
      id: postData.categories.data[0].id,
      displayName: postData.categories.data[0].attributes.displayName,
      slug: postData.categories.data[0].attributes.slug,
    },
  ],
};

// const mockPostData: PostProps = {
//   id: '1',
//   title:
//     'Conheça o DALL-E, inteligência artificial que cria qualquer imagem a partir de uma descrição de texto',
//   excerpt:
//     'Novas tecnologias que misturam linguagem e imagens poderiam servir a artistas gráficos. Preocupação é que acelere também campanhas de desinformação',
//   cover: {
//     id: '1',
//     alternativeText: 'Texto alternativo da capa',
//     url: 'https://s2-oglobo.glbimg.com/CvRnMai0Um4synVbwseVuxGgoxg=/0x0:1024x1024/984x0/smart/filters:strip_icc()/i.s3.glbimg.com/v1/AUTH_da025474c0c44edd99332dddb09cabe8/internal_photos/bs/2022/B/c/E2AX3uSWyBqSzncbwx5A/3.glbimg.com-v1-auth-0ae9f161c1ff459593599b7ffa1a1292-images-escenic-2022-4-13-19-1649888639951.jpg',
//   },
//   content:
//     'SÃO FRANCISCO - No OpenAI, um dos laboratórios de Inteligência Artificial mais ambiciosos do mundo, os pesquisadores estão desenvolvendo tecnologia que permite criar imagens digitais simplesmente descrevendo o que você deseja ver. A OpenAI, apoiada por US$ 1 bilhão em financiamento da Microsoft, ainda não está compartilhando a tecnologia com o público em geral. Mas, recentemente, Alex Nichol, um dos pesquisadores por trás do sistema, demonstrou como ele funciona.Nichol pediu ao sistema “um bule em forma de abacate”, digitando as palavras em uma tela de computador praticamente vazia. Ele criou 10 imagens distintas de um bule de abacate verde-escuro, algumas com caroço e outras sem.',
//   author: {
//     id: '1',
//     displayName: 'Cade Metz',
//     slug: 'autor-do-post',
//   },
//   categories: [
//     {
//       id: '1',
//       displayName: 'Categoria do Post',
//       slug: 'categoria-do-post',
//     },
//   ],
//   createdAt: '2022-02-18',
// };

export default mockPostData;
