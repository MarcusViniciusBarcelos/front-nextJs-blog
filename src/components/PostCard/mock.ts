import { PostCardProps } from '.';
import { data } from '../../api/dados.json';

const postData = data.posts.data[1].attributes;

const mockPostData: PostCardProps = {
  id: data.posts.data[1].id,
  slug: postData.slug,
  title: postData.title,
  excerpt: postData.excerpt,
  cover: postData.cover.data.attributes.url,
};

export default mockPostData;
