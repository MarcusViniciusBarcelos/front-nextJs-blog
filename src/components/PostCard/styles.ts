import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    ${HeadingStyles} {
      margin: 0;
      margin-top: calc(${theme.spacings.small} - 0.6);
      margin-bottom: ${theme.spacings.small};
    }

    h2 {
      transition: all 300ms ease-in-out;
    }

    &:hover h2 {
      color: ${theme.colors.secondary};
    }

    &:hover img {
      opacity: 0.8;
      transform: scale(1.03);
    }
  `}
`;

export const Cover = styled.img`
  max-width: 100%;
  transition: all 300ms ease-in-out;
`;

export const Excerpt = styled.p`
  max-width: 100%;
`;
