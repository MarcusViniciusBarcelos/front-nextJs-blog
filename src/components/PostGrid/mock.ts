import { PostGridProps } from '.';
import { data } from '../../api/dados.json';

const postsData = data.posts.data;

const mockPostGrid: PostGridProps = {
  posts: postsData.map((post) => ({
    id: post.id,
    slug: post.attributes.slug,
    title: post.attributes.title,
    excerpt: post.attributes.excerpt,
    cover: post.attributes.cover.data.attributes.url,
  })),
};

export default mockPostGrid;
