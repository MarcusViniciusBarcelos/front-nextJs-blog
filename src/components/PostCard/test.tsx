import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { PostCard, PostCardProps } from '.';

import mock from './mock';

const props: PostCardProps = mock;

describe('<PostCard />', () => {
  it('should render', () => {
    renderTheme(<PostCard {...props} />);

    expect(screen.getByRole('heading', { name: 'Oi' })).toBeInTheDocument();
  });
});
