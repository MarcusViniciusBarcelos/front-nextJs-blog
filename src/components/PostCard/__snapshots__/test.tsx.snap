// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
.c3 {
  color: #000000;
  line-height: 1.2;
  font-size: 2.4rem;
  text-transform: none;
}

.c0 .c2 {
  margin: 0;
  margin-top: calc(1.6rem - 0.6);
  margin-bottom: 1.6rem;
}

.c0 h2 {
  transition: all 300ms ease-in-out;
}

.c0:hover h2 {
  color: #dc143c;
}

.c0:hover img {
  opacity: 0.8;
  transform: scale(1.03);
}

.c1 {
  max-width: 100%;
  transition: all 300ms ease-in-out;
}

.c4 {
  max-width: 100%;
}

<div>
  <div
    class="c0"
  >
    <a
      href="/post/anunciando-elden-ring-a-cidade-subterranea"
    >
      <img
        alt="Anunciando Elden Ring: A Cidade Subterrânea"
        class="c1"
        src="https://res.cloudinary.com/dm5kvicsz/image/upload/v1708018948/H7_AL_Wt_R_md_9dcdc521e6.jpg"
      />
    </a>
    <a
      href="/post/anunciando-elden-ring-a-cidade-subterranea"
      style="text-decoration: none;"
    >
      <h2
        class="c2 c3"
      >
        Anunciando Elden Ring: A Cidade Subterrânea
      </h2>
    </a>
    <p
      class="c4"
    >
      Prepare-se para explorar um novo mundo em Elden Ring: A Cidade Subterrânea, uma expansão épica que leva você às profundezas das Terras Intermediárias.
    </p>
  </div>
</div>
`;
